src/charm.py

```diff
@@ -66,12 +66,12 @@ def version(self) -> Optional[str]:
 
 
 class AlertmanagerCharm(CharmBase):
-    _container_name: str = "alertmanager"  # automatically determined from charm name
+    _container_name = "alertmanager"  # automatically determined from charm name
     _layer_name = "alertmanager"  # layer label argument for container.add_layer
-    _service_name: str = "alertmanager"  # chosen arbitrarily to match charm name
-    _peer_relation_name: str = "replicas"  # must match metadata.yaml peer role name
-    _api_port: int = 9093  # port to listen on for the web interface and API
-    _ha_port: int = 9094  # port for HA-communication between multiple instances of alertmanager
+    _service_name = "alertmanager"  # chosen arbitrarily to match charm name
+    _peer_relation_name = "replicas"  # must match metadata.yaml peer role name
```

Comment:
You could get this dynamically using the [`CharmMeta`](https://ops.readthedocs.io/en/latest/#ops.charm.CharmMeta) class

Reply:
Same for `_storage_path` below

Reply:
- peers: do you mean `next(iter(self.meta.peers.keys()))` under the assumption there is only one?
- storage: in metadata.yaml, I have `storage:  data: ...`, where "data" is arbitrary, so I'd have to hard code something anyway: `self.meta.storages['data'].location`, unless I assume there is only one. I actually do prefer hard-coding "data" over hard-coding the location itself, but just wanted to make sure what you meant.

Reply:
On peers: yes - it might be prudent to check there is indeed only one peer relation

On storage - indeed, that is what I meant :)

Reply:
Done

Reply:
Is there another push which changes it here?

If not, but can get this dynamically via `next(iter(self.meta.peers)).name` or `list(self.meta.peers.keys())[0].name` or other. I don't think hardcoding it bad, but having a `# must match metadata.yaml peer role name` as a comment may make readers wonder "why don't you just fetch it from there?"

One or the other (hardcoding without the comment) may be better. Similarly, just above this, removing "arbitrarily" is probably a good idea, since that makes it sound as if there was not a design decision behind this, and there was -- consistency. `# Service name matches charm name for consistency` reads better.

Really a nit, though

Reply:
There was a commit with next(iter(...)) but I reverted it after realizing more than one peer relation is possible.